lib/miq_automation_engine/models/miq_ae_domain.rb

```diff
@@ -107,6 +108,25 @@ def available_version
     version_field.try(:fetch_path, 'field', 'default_value')
   end
 
+  def self.import_git_url(options)
+    gr = GitRepository.find_or_create_by(:url => options['url'])
+    if options['userid'] && options['password']
+      auth = gr.authentications.detect { |item| item.authtype == 'default' }
+      if auth
+        auth.update_attributes(options.slice(*AUTH_KEYS))
+      else
+        gr.authentications.create(options.slice(*AUTH_KEYS).merge(:authtype => 'default'))
+      end
+    end
+
+    gr.refresh
+    gr.reload
```

Comment:
curious that a reload is required...refresh should put the gr object in a "reloaded" state, and the this line is not necessary.